wurst/objects/abilities/Omnicure.wurst

```diff
@@ -40,19 +46,11 @@ class Omnicure extends ChannelAbilityPreset
         this.setArtDuration(1, 0)
 
 
-@compiletime function createOmnicure()
-    new Omnicure(ABILITY_OMNICURE, "E", new Pair(1, 0))
+public let OMNICURE = compiletime(new Omnicure(ABILITY_OMNICURE, "E", new Pair(1, 0), MANACOST))
 
 init
-    registerSpellEffectEvent(ABILITY_OMNICURE) ->
-        let heroes = GetUnitsInRectMatching(bj_mapInitialPlayableArea, Condition(() -> begin
-            let filterUnit = GetFilterUnit()
-            return filterUnit.isType(UNIT_TYPE_HERO) and filterUnit.getOwner().isAllyOf(GetSpellAbilityUnit().getOwner())
-        end))
-
-        let owner = GetSpellAbilityUnit().getOwner()
-        for hero in heroes
-            new DummyCaster()
-                ..owner(owner)
-                ..origin(hero.getPos() + vec2(100, 0))
-                ..castTarget(ABILITY_CURE_ALL, 1, Orders.autodispel, hero)
+    EventListener.onCast(ABILITY_OMNICURE) (unit caster) ->
+        let owner = caster.getOwner()
+        forUnitsInRect(bj_mapInitialPlayableArea) (unit u) ->
+            if u.isType(UNIT_TYPE_HERO) and u.getOwner().isAllyOf(owner)
+                InstantDummyCaster.castTarget(owner, ABILITY_CURE_ALL, 1, Orders.autodispel, u)
```

Comment:
This is the wrong order ID. Not sure if you can use the compiletime definition.

Reply:
Wrong order ID? You mean it should be abolishmagic? Funny I didn't edit this dummyCast, but it's still working, gonna change to abolish just to make sure.

Reply:
Well cure all use autodispell and there's no order for abolish magic.

Reply:
Yeah you're right

Reply:
Confused myself. I was thinking this was where the potion was dummy casting the priest omnicure, not the priest omnicure casting cure all.